### PR TITLE
[WFLY-18073] Add the OWASP dependency-check plugin to the WildFly build.

### PR DESCRIPTION
**Pull request by darranl**
I am raising this as a draft whilst I put together some documentation but in the meantime it should be possible to experiment with the plugin.

https://issues.redhat.com/browse/WFLY-18073

The plugin does not run by default, also at the moment it is not possible to run in CI due to rate limiting of the vulnerability database APIs but after the first run subsequent local runs are very fast to generate a report on-demand.

Before running you will need a personal API key from NVD, this can be obtained by submitting the following form:

https://nvd.nist.gov/developers/request-an-api-key

This API key will need adding to your Maven settings.xml:

```
    <server>
        <id>nvd</id>
        <password>XXX</password>
    </server>

```

Run a `mvn install` build of WildFly as normal so that any dependencies are downloaded and the artefacts of the build installed to your local maven repo.

The following command can be used to generate a report:

`mvn -Ddependency-check  dependency-check:aggregate`

The report will be found in `wildfly/target/dependency-check-report.html` which you can open in your local browser.

To suppress false positives entries are added to the new `sca-overrides/owasp-suppressions.xml` file.  There are two approaches which we can use to suppress a report, we can suppress the exact CVE against or a component or we can suppress how the component was matched.

Although vulnerabilities are suppressed it is possible in the report to see these to double check.

A further challenge will be mapping of false negatives, if we identify vulnerabilities in components which are not picked up we will need to triage.

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19022](https://issues.redhat.com/browse/WFLY-19022)
> * [WFLY-19088](https://issues.redhat.com/browse/WFLY-19088)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

**Comment by darranl** [comment]
@bstansberry as this is an independent build profile I have set as ready to review.

File: sca-overrides/owasp-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      file name: jgroups-aws-3.0.0.Final.jar
+      ]]></notes>
+      <packageUrl regex="true">^pkg:maven/org\.jgroups\.aws/jgroups\-aws@.*$</packageUrl>
+      <cve>CVE-2016-2141</cve>
+   </suppress>
+   <suppress until="2025-03-04">
+      <notes><![CDATA[
+      file name: commons-compress-1.24.0.jar
+      ]]></notes>
+      <packageUrl regex="true">^pkg:maven/org\.apache\.commons/commons\-compress@.*$</packageUrl>
```
Comment (bstansberry) [inline sca-overrides/owasp-suppressions.xml:153]: Should this have a comment?

Perhaps "Not used in a WildFly installation"

File: sca-overrides/owasp-suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      ]]></notes>
+      <!-- The regex was adjusted to match all apacheds artefacts, in most other cases we would use the rule as-proposed. -->
+      <packageUrl regex="true">^pkg:maven/org\.apache\.directory\.server/apacheds\-.*@.*$</packageUrl>
+      <cve>CVE-2010-1151</cve>
+   </suppress>
+   <suppress until="2025-02-14">
+      <notes><![CDATA[
+      file name: mina-core-2.1.3.jar
+      ]]></notes>
+      <packageUrl regex="true">^pkg:maven/org\.apache\.mina/mina\-core@.*$</packageUrl>
```
Comment (bstansberry) [inline sca-overrides/owasp-suppressions.xml:109]: Should this have a comment?

File: sca-overrides/owasp-suppressions.xml [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      <packageUrl regex="true">^pkg:maven/com\.fasterxml\.jackson\.core/jackson\-databind@.*$</packageUrl>
+      <!-- Disputed by FasterXML that this is a vulnerability https://github.com/FasterXML/jackson-databind/issues/3972 -->
+      <cve>CVE-2023-35116</cve>
+   </suppress>
+   <suppress until="2025-02-09">
+      <notes><![CDATA[
+      file name: jakarta.security.enterprise-3.0.3.jar
+      ]]></notes>
+      <packageUrl regex="true">^pkg:maven/org\.glassfish\.soteria/jakarta\.security\.enterprise@.*$</packageUrl>
+      <vulnerabilityName>CVE-2020-1732</vulnerabilityName>
```
Comment (bstansberry) [inline sca-overrides/owasp-suppressions.xml:140]: Should this have a comment?

**Review by bstansberry — COMMENTED** [review]
@darranl This LGTM except some of the suppressions could use a comment, if possible.

I ignored some that don't have a comment where the CVE is before 2020

**Comment by darranl** [comment]
@bstansberry before you add too many "Should this have a comment?" comments - this is why you see in the PR quite a few commits - git annotated can be used on this file to identify why each block was added.

But can add comments as well if you prefer.

**Review by bstansberry — APPROVED** [review]
I think further comments can be added later.

**Comment by bstansberry** [comment]
Thanks @darranl

**Pull request merged**